src/dc_receive_imf.rs

```diff
@@ -827,37 +827,73 @@ async fn add_parts(
 
     // Apply ephemeral timer changes to the chat.
     //
-    // Only non-hidden timers are applied now. Timers from hidden
+    // Only non-hidden timers are applied. Timers from hidden
     // messages such as read receipts can be useful to detect
     // ephemeral timer support, but timer changes without visible
     // received messages may be confusing to the user.
     if !*hidden
         && !location_kml_is
         && !is_mdn
-        && (is_dc_message != MessengerMessage::Yes
-            || parent.is_none()
-            || parent.unwrap().ephemeral_timer != ephemeral_timer)
         && chat_id.get_ephemeral_timer(context).await? != ephemeral_timer
-        && chat_id
-            .update_timestamp(context, Param::EphemeralSettingsTimestamp, *sent_timestamp)
-            .await?
     {
-        if let Err(err) = chat_id
-            .inner_set_ephemeral_timer(context, ephemeral_timer)
-            .await
+        info!(
+            context,
+            "received new ephemeral timer value {:?} for chat {}, checking if it should be applied",
+            ephemeral_timer,
+            chat_id
+        );
+        if is_dc_message == MessengerMessage::Yes
+            && mime_parser
+                .parts
+                .get(0)
+                .and_then(|part| part.param.get(Param::Quote))
+                .is_none()
+            && parent.map(|p| p.ephemeral_timer) == Some(ephemeral_timer)
```

Comment:
note to myself: `parent.map(|p| p.ephemeral_timer) == Some(ephemeral_timer)` correspond to old ` parent.is_none()
             || parent.unwrap().ephemeral_timer != ephemeral_timer)` while the Quote-check is complely new.

was first confused as i thought the quote-check substitues the parent check somehow :)